apps/srp.c

```diff
@@ -151,14 +153,16 @@ static char *srp_create_user(char *user, char **srp_verifier,
                              char **srp_usersalt, char *g, char *N,
                              char *passout, int verbose)
 {
-    char password[1024];
+    char password[1025];
     PW_CB_DATA cb_tmp;
     char *gNid = NULL;
     char *salt = NULL;
+    int len;
     cb_tmp.prompt_info = user;
     cb_tmp.password = passout;
 
-    if (password_callback(password, sizeof(password), 1, &cb_tmp) > 0) {
+    if ((len = password_callback(password, sizeof(password)-1, 1, &cb_tmp)) > 0) {
+        password[len] = 0;
```

Comment:
I also just pushed a commit to add OPENSSL_cleanse(passwd,len) as suggested.

As for adding OPENSSL_clear_free(pass{in|out}) there is no app which uses the app_passwd() function that uses OPENSSL_clear_free() or OPENSSL_cleanse() on the result and anyhow there are quite a few strdup()'s and the like in app_passwd() itself, the password data coming from a file, fd, environment or even CLI argument, so I do think it is worth it.

However, while looking at it I realized that the srp app does print out the password to stderr unconditionally when verifying it, which looks like bad practice to me. Should I remove that? Or at least make it conditional on a high verbose level?